scripts/toolchains/osx.cmake

```
@@ -2,17 +2,26 @@ if(NOT _VCPKG_OSX_TOOLCHAIN)
 set(_VCPKG_OSX_TOOLCHAIN 1)
 if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
     set(CMAKE_CROSSCOMPILING OFF CACHE BOOL "")
-
     set(CMAKE_SYSTEM_VERSION "${CMAKE_HOST_SYSTEM_VERSION}" CACHE STRING "")
-    set(CMAKE_SYSTEM_PROCESSOR "${CMAKE_HOST_SYSTEM_PROCESSOR}" CACHE STRING "")
 else()
     set(CMAKE_SYSTEM_VERSION "17.0.0" CACHE STRING "")
-    set(CMAKE_SYSTEM_PROCESSOR "x86_64" CACHE STRING "")
 endif()
 set(CMAKE_SYSTEM_NAME Darwin CACHE STRING "")
 
 set(CMAKE_MACOSX_RPATH ON CACHE BOOL "")
 
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+   set(CMAKE_SYSTEM_PROCESSOR x86_64 CACHE STRING "")
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+   set(CMAKE_SYSTEM_PROCESSOR x86 CACHE STRING "")
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+   set(CMAKE_SYSTEM_PROCESSOR arm64 CACHE STRING "")
+elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
+   set(CMAKE_SYSTEM_PROCESSOR "${CMAKE_HOST_SYSTEM_PROCESSOR}" CACHE STRING "")
+else()
+   set(CMAKE_SYSTEM_PROCESSOR x86_64 CACHE STRING "")

```

This is an error; we do not support anything other than x86_64 and ARM64 on macOS (sorry powerbook fans), and `VCPKG_TARGET_ARCHITECTURE` must be set correctly

```suggestion
else()
    message(FATAL_ERROR "Unrecognized VCPKG_TARGET_ARCHITECTURE for macOS - ${VCPKG_TARGET_ARCHITECTURE}")
```

---

Building fails with this change, see: 
https://github.com/microsoft/vcpkg/pull/21790#issuecomment-1005210085